### PR TITLE
chore(RHTAPWATCH-564): Run exporter service Go unit tests for each PR opened

### PR DESCRIPTION
**Pull request by Kousalya1998**
- Add GO unit tests on each opened pull request.

Ticket:
[RHTAPWATCH-564](https://issues.redhat.com/browse/RHTAPWATCH-564)

### CODE REVIEW

File: .github/workflows/unit_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+jobs:
+  go-test:
+    runs-on: ubuntu-latest
+    container:
+      image: registry.access.redhat.com/ubi8/go-toolset:1.19.13-2.1698062273
```
Comment (amisstea) [inline .github/workflows/unit_tests.yaml:13]: Shouldn't we be using go 1.20.10+ at this point to mitigate CVE-2023-44487? See [RHTAPWATCH-546](https://issues.redhat.com/browse/RHTAPWATCH-546) which predated this new exporter. cc @hmariset

Comment (hmariset) [reply .github/workflows/unit_tests.yaml:13]: yes, `registry.access.redhat.com/ubi9/go-toolset:1.20.10-2` is the image we are using now in segment-bridge

Comment (Kousalya1998) [reply .github/workflows/unit_tests.yaml]: Updated

File: .github/workflows/unit_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          path: ~/packages
+          key: jq-1.7-${{ runner.os }}
+        if: success()
+      - if: ${{ steps.cache-jq.outputs.cache-hit != 'true' }}
+        name: Install jq
+        run: |
+          mkdir -p ~/packages && \
+          curl -L \
+          https://github.com/stedolan/jq/releases/download/jq-1.7/jq-linux64 \
```
Comment (Omeramsc) [inline .github/workflows/unit_tests.yaml:29]: Is there a need of jq in this repo's unit tests?

**Review by Omeramsc — CHANGES_REQUESTED** [review]
Isn't Tekton being mostly used in the o11y repo?

**Comment by yftacherzog** [comment]
> Isn't Tekton being mostly used in the o11y repo?

I'd also prefer we stick with tekton unless we have a compelling reason for introducing another flow. The ticket was also mentioning tekton for that reason.

**Comment by Kousalya1998** [comment]
> Isn't Tekton being mostly used in the o11y repo?

With Go unit tests, I followed the same approach as segment-bridge and used GitHub actions. If o11y uses only tekton then it makes sense to have only tekton.

**Review by amisstea — APPROVED** [review]

**Comment by hmariset** [comment]
LGTM

**Review by hmariset — APPROVED** [review]

**Review by amisstea — APPROVED** [review]

**Comment by amisstea** [comment]
@Omeramsc's concern was addressed. Merging this.

**Pull request merged**